EcoSoapBank/Model/ImpactStats.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  ImpactStats.swift
+//  EcoSoapBank
+//
+//  Created by Shawn Gee on 8/11/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import UIKit
+
+struct ImpactStats: Decodable {
+    
+    let allStats: [Statistic]
```

Comment:
Hmmm... so the way this is currently set up, this array could technically be created to have multiple instances of each statistic (eg, it could have both `soapRecycled(grams: 20)` and `soapRecycled(grams: 30)`. I really like using enums with associated values, but I wonder it would be better in this case to just have a struct with a property for each statistic? e.g.:

```swift
struct ImpactStats {
    let soapRecycled: Int
    let linensRecycled: Int
    let bottlesRecycled: Int
    let paperRecycled: Int
    let peopleServed: Int
    let womenEmployed: Int

    // etc
}
```

I think since we're never directly creating it, it shouldn't be a problem, but it might make the codebase simpler, and it means we're using the compiler to enforce that we don't get into an "impossible" state rather than relying on the logic flow we write.

Reply:
I started out with a simple struct with Ints for the statistics but was struggling to come up with a good way of associating the description, units, and image with each statistic in order to populate the cells. I can compromise and store each Statistic separately and have allStatistics return an array with all of the stats in it. Or if you have a better suggestion?